generator/dart/packages/google_cloud_common/lib/common.dart

```diff
@@ -91,7 +93,49 @@ abstract class CloudService {
       return body.isEmpty ? {} : jsonDecode(body);
     }
 
-    // TODO(#1454): This is a placeholder until we're able to parse 'Status'.
-    throw ClientException('${response.statusCode}: ${response.reasonPhrase}');
+    Status status;
+
+    try {
+      final json = jsonDecode(response.body);
+      status = Status.fromJson(json['error']);
+    } catch (_) {
+      // If we're not able to parse the Status error, return a general HTTP
+      // exception.
+      throw ClientException('${response.statusCode}: ${response.reasonPhrase}');
+    }
+
+    throw status;
   }
 }
+
+T? $toMessage<T>(dynamic json, T Function(Map<String, dynamic>) decoder) {
```

Comment:
For my education, what does the `$` mean here?

Reply:
It's just a regular identifier character. it's almost never used in hand-written code, but is sometimes used in generated code to create terse symbols that are unlikely to collide w/ other things in the library. For example:

```dart
import 'package:foo/bar.dart' as $0;

...

$0.FooBarClass.doSomething();
```

creates an import prefix that likely won't conflict w/ anything in that file.